### DISCUSSION TITLE
动态切换快捷键方案

### DISCUSSION THREAD

**Discussion by hooke007** [hooke007/mpv_PlayKit#415] [INFO 信息]
简单讲，可以在运行时切换使用不同的 "input.conf" 文件。

需要此节点之后的mpv版本才支持这个特性 https://github.com/mpv-player/mpv/commit/4ab521f08010329701885d4e383a6de77589a389

曾经用过的不少（客制化）键盘有一个“切层”的功能 https://ydkb.io/
虽然只有一把键盘，但是依靠按某个键位，就可以把当前的层切换（每一层对于的按键功能也不同）实现一把键盘干多个键盘的事情。

提到这个事情是因为mpv最近引入的命令 [load-input-conf](https://mpv.io/manual/master/#command-interface-load-input-conf) 让我联想到了有点近似的它。

mpv因为自身的特性也被不少人用来做音乐播放器和图片查看器，在“音乐模式”或“图片模式”时需要使用相同按键的不同的快捷键功能，
这也是我能想到的其它比较现实的应用场景。

### 仿键盘切层示例
（如果你用过切层键盘，这个例子应该很形象，没有的话就可能十分抽象。。。）
以空配置目录开始，新建 input.conf 和 input2.conf 文件（分别对应层1，层2）
分别写入
```
F1   show-text "当前已是1号快捷键方案"
F2   load-input-conf "~~/input2.conf" ; show-text "切换2号快捷键方案"
1   add volume -1
2   add volume 1
```
```
F1   load-input-conf "~~/input.conf" ; show-text "切换1号快捷键方案"
F2   show-text "当前已是2号快捷键方案"
1   add speed -0.1
2   add speed 0.1
```
然后运行mpv，因为没有其它设置，所以默认加载的是“1号快捷键方案”，此时按<kbd>1</kbd><kbd>2</kbd>的功能就是调节音量。
现在你可以按<kbd>F2</kbd>加载“input2.conf”（同时input.conf被自动卸载），此时再按<kbd>1</kbd><kbd>2</kbd>的功能就是调节速度。
当然也可以再按<kbd>F1</kbd>重载“input.conf”

**Comment by wathper** [comment]
在使用了`dyn_menu.lua`菜单脚本后，动态切换到其他没有正确配置`dyn_menu.lua`语法的快捷键方案、并且激活了上下文菜单时，有概率导致`dyn_menu.lua`脚本失效或者崩溃。
需要设置

`MBTN_Right       load-input-conf "~~/input.conf" ; script-message-to dyn_menu show`

在激活上下文菜单之前切换回默认的快捷键方案。
